units/stats_host_idle.go

```diff
@@ -95,26 +108,31 @@ func (j *collectHostIdleDataJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
-	settings := j.env.Settings()
+	if j.settings == nil {
```

Comment:
Why is this needed? The Settings() function won't load them if it's nil (and it shouldn't be nil if you got to here)

Reply:
There are three ways this job can be constructed: 
1. Directly via the public `NewCollectHostIdleDataJob` (j.settings is nil,) 
2. Directly via the private constructor (when called directly from another job: j.settings is populated.) 
3. Via the factory in the job registry, (j.settings is nill and cannot be populated.) 

So we have to be careful to populate the attribute when we execute the job.

Reply:
But the function will just return nil if if settings is nil which is fine. Also this should be j.env.settings if you're going to check that

Reply:
the case I'm protecting against is the case where the job is instantiated without a settings object (which I think is fairly common) and we need to pull it from the environment. 

In one major case, the object will be instantiated with a settings object, and in those cases, we shouldn't re-requery.